Src/Base/AMReX_FabArrayBase.cpp

```diff
@@ -2484,6 +2484,30 @@ FabArrayBase::is_cell_centered () const noexcept
     return boxArray().ixType().cellCentered();
 }
 
+bool
+FabArrayBase::isFusingCandidate () const noexcept
+{
+#ifdef AMREX_USE_GPU
+    // This is fine tuned on MI100.
+    // For V100 and A100, it is not very sensitive to the choice here.
+    const int n = local_size();
+    if (n <= 1) {
+        return false;
+    } else if (n > 8) {
+        return true;
+    } else {
+        for (int i = 0; i < n; ++i) {
+            if (boxarray[indexArray[i]].numPts() <= Long(65*65*65)) {
+                return true;
+            }
+        }
+        return false;
+    }
+#else
+    return false;
+#endif
+}
+
```

Comment:
Would it make sense to right now setup to allow different versions for different build types? i.e:
`#ifdef AMREX_USE_CUDA
...
#elif defined(AMREX_USE_HIP)`
or even do it for individual hardware? I assume MI100 and MI200 might have very different optimizations here (although we might not care about MI100 at all once we get MI200s).

Reply:
Yes, in the future, we might use macros that depend on hardware and software (e.g., rocm versions).  For V100 and A100, the choice here does not appear to matter much.  We could even simply return true for all cases.

Reply:
Really? Even for 2 128^3's? That's impressive. Sounds like something we should get the AMD folks looking at. Is there a "fused" benchmark?

Reply:
For the test here, https://github.com/WeiqunZhang/amrex-devtests/blob/main/mfparfor_test/main.cpp using https://github.com/WeiqunZhang/amrex/tree/fuse_mfutil, on A100 I get,
```
$ ./main*.ex box_sizes="128" nboxes="2"
...
    Kernel run time is 1.8165400000e-04.
```
If I make the following changes to amrex
```
--- a/Src/Base/AMReX_FabArrayBase.cpp
+++ b/Src/Base/AMReX_FabArrayBase.cpp
@@ -2487,6 +2487,7 @@ FabArrayBase::is_cell_centered () const noexcept
 bool
 FabArrayBase::isFusingCandidate () const noexcept
 {
+    return true;
 #ifdef AMREX_USE_GPU
     // This is fine tuned on MI100.
     // For V100 and A100, it is not very sensitive to the choice here.
```
I get
```
$ ./main*.ex box_sizes="128" nboxes="2"
...
    Kernel run time is 1.7696400000e-04.
```

Reply:
Huh. Interesting. Not what I would expect. I wonder if the A100s have enough registers that both of the boxes run at the same time, reducing the additional fusing overheads, maybe?